CryptoSwiftTests/CryptoSalsa20Test.swift

```diff
@@ -0,0 +1,346 @@
+            XCTAssertEqual(vector.stream2, Array(pb[vector.stream2Pos]), "Test: \(vector.name) - Stream 2 does not Equals")
+            XCTAssertEqual(vector.stream3, Array(pb[vector.stream3Pos]), "Test: \(vector.name) - Stream 3 does not Equals")
+            XCTAssertEqual(vector.stream4, Array(pb[vector.stream4Pos]), "Test: \(vector.name) - Stream 4 does not Equals")
+            
+            //build XOR of complete cipher block
+            let xorDigits = self.generateXorDigits(pb, blockSize: vector.xorDigits.count)
+            XCTAssertEqual(vector.xorDigits, xorDigits, "Test: \(vector.name) - XOR-Digits does not Equals")
+        }
+    }
+    
+    /**
+     Run a performance test
+     */
+    func testSalsa20Performance() {
+        let key:[UInt8] = [0x2b,0x7e,0x15,0x16,0x28,0xae,0xd2,0xa6,0xab,0xf7,0x15,0x88,0x09,0xcf,0x4f,0x3c];
+        let iv:[UInt8] = [0x00,0x01,0x02,0x03,0x04,0x05,0x06,0x07,0x08,0x09,0x0A,0x0B,0x0C,0x0D,0x0E,0x0F]
+        let message = [UInt8](count: (1024 * 1024) * 1, repeatedValue: 7)
+        measureMetrics([XCTPerformanceMetric_WallClockTime], automaticallyStartMeasuring: true, forBlock: { () -> Void in
+            let encrypted = try! Salsa20(key: key, iv: iv)?.encrypt(message)
+            self.stopMeasuring()
+            XCTAssert(encrypted != nil, "not encrypted")
+        })
+    }
+    
+    /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+     * Helper Functions
+     * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+    
+    /**
+     Run Sasla20 cipher some rounds with random buffer sizes to a defined position
+     
+     - parameter c:          The Salsa20 cipher
+     - parameter nPos:       Define the start position
+     - parameter nTargetPos: Define the end position
+     
+     - returns: return the end defnied position
+     */
+    internal static func salsa20ToPos(c: Salsa20, nPos: Int, nTargetPos: Int) -> Int {
+        var tmp = nPos
+        //run until defined positon is reached
+        while(tmp < nTargetPos) {
+            //create a random buffer size
+            let x = Int(arc4random_uniform(513))
+            //check if random buffer size is greater then defined position, and cut it off to defined position
+            let nGen = min(nTargetPos - tmp, x)
+            //create buffer of random size with 0 filles
+            var pb = [UInt8](count:nGen, repeatedValue: 0)
+            //encrypt and run Salsa20 cipher to the defined position
+            try! pb = c.encrypt(pb)
+            //save run size
+            tmp += nGen
+        }
+        return nTargetPos
+    }
+
+    /**
+     Helper function which load the Test-Vector-Data from a File
+     
+     - returns: Returns the data as an array of SalsaTestVector
+     */
+    private func loadTestVectors() -> [SalsaTestVector] {
```

Comment:
ok, can you think about make it simpler? By simpler here I mean simpler to maintain by not building parser for text file to get binary representation of test vectors. I'm fine with inlined arrays as well, nothing wrong with that.